ibm/service/power/resource_ibm_pi_route.go

```diff
@@ -0,0 +1,333 @@
+
+	body := &models.RouteCreate{
+		Action:          &action,
+		Advertise:       &advertise,
+		Destination:     &destination,
+		DestinationType: &destinationType,
+		Enabled:         &enabled,
+		Name:            &name,
+		NextHop:         &nextHop,
+		NextHopType:     &nextHopType,
+	}
+
+	if v, ok := d.GetOk(Arg_UserTags); ok {
+		userTags := flex.FlattenSet(v.(*schema.Set))
+		body.UserTags = userTags
+	}
+
+	route, err := routeClient.Create(body)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	if _, ok := d.GetOk(Arg_UserTags); ok {
+		if route.Crn != nil {
+			oldList, newList := d.GetChange(Arg_UserTags)
+			err := flex.UpdateGlobalTagsUsingCRN(oldList, newList, meta, string(*route.Crn), "", UserTagType)
+			if err != nil {
+				log.Printf("Error on update of pi network security group (%s) pi_user_tags during creation: %s", *route.ID, err)
+			}
+		}
+	}
+
+	routeID := *route.ID
+	d.SetId(fmt.Sprintf("%s/%s", cloudInstanceID, routeID))
+
+	return resourceIBMPIRouteRead(ctx, d, meta)
+}
+
+func resourceIBMPIRouteRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sess, err := meta.(conns.ClientSession).IBMPISession()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	cloudInstanceID, routeID, err := splitID(d.Id())
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	routeClient := instance.NewIBMPIRouteClient(ctx, sess, cloudInstanceID)
+	route, err := routeClient.Get(routeID)
+	if err != nil {
+		if strings.Contains(strings.ToLower(err.Error()), NotFound) {
+			d.SetId("")
+			return nil
+		}
+		return diag.FromErr(err)
+	}
+	d.Set(Arg_CloudInstanceID, cloudInstanceID)
+	d.Set(Arg_Action, route.Action)
+	d.Set(Arg_Advertise, route.Advertise)
+	if route.Crn != nil {
+		d.Set(Attr_CRN, route.Crn)
+		tags, err := flex.GetGlobalTagsUsingCRN(meta, string(*route.Crn), "", UserTagType)
+		if err != nil {
+			log.Printf("Error on get of pi route (%s) pi_user_tags: %s", *route.ID, err)
+		}
+		d.Set(Arg_UserTags, tags)
+	}
+	d.Set(Arg_Destination, route.Destination)
+	d.Set(Arg_DestinationType, route.DestinationType)
+	d.Set(Arg_Enabled, route.Enabled)
+	d.Set(Attr_RouteID, route.ID)
+	d.Set(Arg_Name, route.Name)
+	d.Set(Arg_NextHop, route.NextHop)
+	d.Set(Arg_NextHopType, route.NextHopType)
+	d.Set(Attr_State, route.State)
+
+	return nil
+}
+
+func resourceIBMPIRouteUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sess, err := meta.(conns.ClientSession).IBMPISession()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	cloudInstanceID, routeID, err := splitID(d.Id())
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	routeClient := instance.NewIBMPIRouteClient(ctx, sess, cloudInstanceID)
+	destination := d.Get(Arg_Destination).(string)
+	name := d.Get(Arg_Name).(string)
+	nextHop := d.Get(Arg_NextHop).(string)
+	body := &models.RouteUpdate{
+		Destination: destination,
+		Name:        name,
+		NextHop:     nextHop,
+	}
```

Comment:
Looking at the swagger, routeUpdate does not have those a required.  We might have to update them only when they have changed